media-plugins/gst-plugins-soup/gst-plugins-soup-1.22.1.ebuild

```diff
@@ -0,0 +1,13 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+GST_ORG_MODULE=gst-plugins-good
+
+inherit gstreamer-meson
+
+DESCRIPTION="HTTP client source/sink plugin for GStreamer"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86 ~x64-macos"
+
+RDEPEND=">=net-libs/libsoup-2.48:2.4[${MULTILIB_USEDEP}]"
```

Comment:
I am not sure how to handle it best, but I think at least transitionally we should have
```
RDEPEND=">=net-libs/libsoup-2.48:2.4[${MULTILIB_USEDEP}] net-libs/libsoup:3.0[${MULTILIB_USEDEP}]"
```
(formatted more proper - I can't do tab in github comments quickly).

It checks at runtime which libsoup has been linked in and uses the same one if any have. To guarantee that to actually work for all possible consumers, we need to have it built for both.
Depending on the details, maybe we can get rid of the 2.4 dep later, IF it builds 2.4 support based on 3.0 headers somehow too - haven't freshly checked the plugin implementation details on that. But for now, I think just depending on both is going to be good enough for now.

Reply:
This change should probably go into a gst-plugins-1.20 revbump as well and possibly stabled sooner, which is something for outside this PR though.